src/test/optimizer/strategy/or_to_union_rule_test.cpp

```diff
@@ -0,0 +1,223 @@
+#include "gtest/gtest.h"
+
+#include "logical_query_plan/union_node.hpp"
+#include "optimizer/strategy/or_to_union_rule.hpp"
+#include "strategy_base_test.hpp"
+#include "testing_assert.hpp"
+
+namespace opossum {
+
+class OrToUnionRuleTest : public StrategyBaseTest {
+ public:
+  void SetUp() override {
+    node_a = MockNode::make(
+        MockNode::ColumnDefinitions{{DataType::Int, "a"}, {DataType::Int, "b"}, {DataType::Int, "c"}}, "a");
+    a_a = node_a->get_column("a");
+    a_b = node_a->get_column("b");
+
+    node_b = MockNode::make(MockNode::ColumnDefinitions{{DataType::Int, "a"}, {DataType::Int, "b"}}, "b");
+    b_a = node_b->get_column("a");
+    b_b = node_b->get_column("b");
+
+    node_c = MockNode::make(
+        MockNode::ColumnDefinitions{{DataType::Int, "a"}, {DataType::Int, "b"}, {DataType::Int, "c"}}, "c");
+    c_a = node_c->get_column("a");
+    c_b = node_c->get_column("b");
+
+    customer_demographics =
```

Comment:
adding "real world tables" here just makes the unit test needlessly complicated imo. Identify  the complicated aspects of TPCDS35 and create properly named tests from them.

The OptimizeTPCDS35 tests is a lot of code, but in fact it doesn't exercise the code more than TwoExistsToUnion.